### PR TITLE
hw-35 added solution

### PR DESCRIPTION
**Pull request by OlesiaKapustynska**

### CODE REVIEW

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java [hunk 1]
```diff
@@ -12,11 +19,46 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
...
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
```
Comment (nikdagaev) [inline src/main/java/ma/hibernate/dao/PhoneDaoImpl.java:46]: you may give this query a more specific naming  (like findAllQuery)

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java [hunk 2]
```diff
@@ -12,11 +19,46 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
...
+            Root<Phone> phoneRoot = query.from(Phone.class);
+            List<Predicate> predicatesList = new LinkedList<>();
```
Comment (nikdagaev) [inline src/main/java/ma/hibernate/dao/PhoneDaoImpl.java:48]: Intresting solution using List)

**Review by nikdagaev — CHANGES_REQUESTED** [review]
good job)

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java (hunk 1 above)
Comment (OlesiaSmitiukh) [reply src/main/java/ma/hibernate/dao/PhoneDaoImpl.java]: I renamed ✔️

**Review by nikdagaev — APPROVED** [review]

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java [hunk 3]
```diff
@@ -12,11 +19,46 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
...
+                for (String parameters : entry.getValue()) {
+                    predicate.value(parameters);
+                }
+                predicatesList.add(predicate);
+            }
+            Predicate[] predicates = predicatesList.stream().toArray(Predicate[]::new);
+            queryByParameters.where(criteriaBuilder.and(predicates));
+            return session.createQuery(queryByParameters).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find phone by this parameters", e);
```
Comment (KseniiaMakarova) [inline src/main/java/ma/hibernate/dao/PhoneDaoImpl.java:61]:
```suggestion
            throw new RuntimeException("Can't find phone by these parameters", e);
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java (hunk 3 above)
Comment (OlesiaSmitiukh) [reply src/main/java/ma/hibernate/dao/PhoneDaoImpl.java]: done ✔️

**Pull request closed without merging**